autosklearn/ensemble_builder.py

```diff
@@ -204,6 +206,24 @@ def __init__(
 
         self.validation_performance_ = np.inf
 
+        # Track the ensemble performance
+        self.datamanager = self.backend.load_datamanager()
+        self.y_valid = self.datamanager.data.get('Y_valid')
+        self.y_test = self.datamanager.data.get('Y_test')
+
+        # Support for tracking the performance across time
+        # A Queue is needed to handle multiprocessing, not only
+        # internally for pynisher calls, but to return data
+        # to the main process
+        # We need to do a put/get to properly initialize the queue
+        # else we run into pynisher problems. For that reason we initialize
+        # a queue namespace. This makes the code robust against a termination
+        mgr = multiprocessing.Manager()
```

Comment:
We're using a regular queue over here inside the pynisher: https://github.com/automl/auto-sklearn/blob/master/autosklearn/evaluation/__init__.py#L221

Do you think that might have the same problem?

Could you please also describe the problems in more detail?

Reply:
I think the use case of L221 mentioned above is different because you actually give the queue to pynisher right?

Here the situation is that you create a queue object in process A (automl), then process B (ensemble) uses it, but also process B launches process C (pynisher).

**The problems in more detail (if you remove this lines):**
If you remove the lines mentioned above, this happens when running the holdout example:
Unhandled exception in thread started by <bound method Thread._bootstrap of <Thread(QueueFeederThread, initial daemon)>>
Traceback (most recent call last):
  File "/home/chico/anaconda3/lib/python3.7/threading.py", line 890, in _bootstrap
**MemoryError**

You can also follow the recommendations here (OMP_NUM_THREADS set to 1 ) instead of this
https://github.com/automl/auto-sklearn/issues/604

Reply:
Interesting. Could you then please update the comment to state that this is necessary because the queue is used by three different processes?

Reply:
Understood, I will add this comment